helper/pdfGeneration.js

```diff
@@ -309,8 +338,333 @@ exports.pdfGeneration = async function pdfGeneration(instaRes) {
+                                            //push the instance questions to the array
+                                             if (anwerObject.responseType == "text" || anwerObject.responseType == "date" || anwerObject.responseType == "number" || anwerObject.responseType == "slider") {
+                                                obj.data.push(anwerObject);
+                                            }else{
+                                                //need to add more code to handle chart data for multiselect/radio
+                                                obj.data.push(anwerObject);
+                                            }
+                                        }
+                                        ++instanceNumber;
+                                    }
+                                    matrixData.push(obj);
+                                }
+                            }));
+
+                           // console.log(arrOfData,'arrOfData')
+                            const result = arrOfData
+                            .map(obj => {
+                                if(obj.responseType == 'matrix'){
+                                    obj.answers = obj.instanceQuestions;
+                                }
+                                return obj;
+                            })
+
+                            arrOfData.forEach((questionObj)=>{
+                               // console.log(questionObj,'questionOb')
+                                if(questionObj.responseType == 'multiselect' || questionObj.responseType == 'radio' ){
+                                    processMatrixResponse(questionObj)
+                                }else if(questionObj.responseType == 'matrix'){
+
+                                    for(let i=0;i<questionObj.instanceQuestions.length;i++){
+
+                                        let eachInstanceAnswerObject = questionObj.instanceQuestions[i];
+
+                                        for(let key in eachInstanceAnswerObject){
+                                            if(key !== 'instanceIdentifier'){
+                                                let answerObject = eachInstanceAnswerObject[key];
+
+                                                if(answerObject.responseType == 'multiselect' || answerObject.responseType == 'radio' ){
+                                                    let result = processMatrixResponse(answerObject)
+                                                    eachInstanceAnswerObject[key] = result;
+                                                }
+
+                                            } 
+                                        }
+
+
+                                    }
+
+                                }
+                            })
+
+                            arrOfData.filter(obj => obj.responseType === "matrix").forEach((questionObj)=>{
```

Comment:
@borkarsaish65 Add comments in this function pls

Reply:
done